src/VisualStudio/CSharp/Impl/CodeModel/CSharpCodeModelService.cs

```diff
@@ -3021,6 +3021,14 @@ protected override ITypeSymbol GetTypeSymbolFromPartialName(string partialName,
 
         public override ITypeSymbol GetTypeSymbolFromFullName(string fullName, Compilation compilation)
         {
+            // Some clients pass us an assembly qualified name, which we don't handle.  Just lop off the
+            // assembly info if so.
+            var index = fullName.IndexOf(',');
+            if (index > 0)
+            {
+                fullName = fullName.Substring(0, index);
+            }
+
             ITypeSymbol typeSymbol = compilation.GetTypeByMetadataName(fullName);
 
             if (typeSymbol == null)
```

Comment:
Note: The call to `Compilation.GetTypeByMetadataName` already existed and was failing, because no, it doesn't handle assembly qualified names.  That led us to go into the fallback case, but ParseTypeName would only parse up to the +, which means that we'd look up the containing type, not the nested type.